### PR TITLE
add `ok` field to token-auth-fails

### PR DESCRIPTION
**Pull request by abdulfaizp**
`ok` field added to decorator response

### CODE REVIEW

**Comment by xoneco** [comment]
Just to post this somewhere offline:

- Still needs response `Bad Request` in response type.

**Comment by tiferrei** [comment]
Just for the record: Partially fixes issue #52

File: backend/uclapi/roombookings/token_auth.py [hunk 1]
```diff
@@ -9,18 +8,22 @@ def wrapped(request, *args, **kwargs):
         token = request.GET.get("token")
 
         if not token:
-            return JsonResponse({
+            response = JsonResponse({
                 "ok": False,
                 "error": "No token provided"
             })
+            response.status_code = 400
```
Comment (tiferrei) [inline backend/uclapi/roombookings/token_auth.py]: I think this should a `401` 👉  [Internet Engineering Task Force (IETF)](https://tools.ietf.org/html/rfc7235#section-3.1)

**Comment by xoneco** [comment]
This should be good to go.

@jermenkoo / @hennersz could you CR and merge, pls?

File: backend/uclapi/roombookings/token_auth.py [hunk 2]
```diff
@@ -9,16 +8,22 @@ def wrapped(request, *args, **kwargs):
...
 
         try:
             App.objects.get(api_token=token)
         except ObjectDoesNotExist:
-            return JsonResponse({
+            response = JsonResponse({
+                "ok": False,
                 "error": "Token does not exist"
             })
+            response.status_code = 400
```
Comment (tiferrei) [inline backend/uclapi/roombookings/token_auth.py]: I think this should a `401` 👉  [Internet Engineering Task Force (IETF)](https://tools.ietf.org/html/rfc7235#section-3.1)

**Comment by xoneco** [comment]
The returning JSON basically takes over the function of being descriptive. Just reading through that section you linked, we don't fulfill like the majority of things they mention (Authorization header, WWW-Authenticate, etc.)

We also use `400` in various other parts of the API, so we should keep it like that for now for consistency. I'm open to changing this API-wide, but that should be done in comprehensive way in a separate PR.

**Comment by ChristopherHammond13** [comment]
Are we good to merge this? If so I'll get it straight into prod now. My only concern is if this constitutes a _breaking_ change; if so, we ought to version this change. @xoneco @abdulfaizp

**Comment by xoneco** [comment]
No, this is not a breaking change. It changes the API to behave the way it is advertised

**Comment by ChristopherHammond13** [comment]
Merged :)

**Pull request merged**